docs/content/uijson_format/index.rst

```diff
@@ -4,31 +4,100 @@ UI.JSON Format
+To complete the example above with a python script to print out the name of a block model object, we will add the **run_command**, and **title** parameter to the ui.json file.
+
+.. code-block:: json
+
+    {
+    "grid_object": {
+    "meshType": ["{B020A277-90E2-4CD7-84D6-612EE3F25051}"],
+    "main": true,
+    "label": "Select Block Model",
+    "value": ""
+    },
+    "title":"My first UI",
+    "run_command": "run_me"
+    }
+
+.. figure:: ./images/block_model_uijson.png
+
+Within the accompanying python script, the parameters from the ui.json is passed to the python script as a dictionary of arguments, and can be accessed using the InputFile module of geoh5py as shown below:
+
+.. code-block:: python
+
+    import sys
+    from geoh5py.ui_json import InputFile
+
+    ui_json = sys.argv[1]
+    ifile = InputFile.read_ui_json(ui_json)
+    params = ifile.data
+
+    # Get the block model grid object
+    bm = params["grid_object"]
+    print(f"The selected object name is {bm.name}")
+
+
+.. figure:: ./images/block_model_output.png
+
+When a **ui.json** is run within Geoscience ANALYST Pro, the following parameters are updated or added:
+
+- The **value** member is updated with the UUID of the object selected in the UI
+- The **enabled** member ``bool`` is set for whether the parameter is enabled
+- The :ref:`Data parameter <data_parameter>` will also have updated **isValue** and **property** members. The **isValue** ``bool`` member is *true* if the **value** member was selected and *false* if the **property** member was selected.
+
+The following JSON objects will be written (and overwritten if given) upon running a ui.json from Geoscience ANALYST Pro:
+
+- **monitoring_directory** ``str`` the absolute path of a monitoring directory. Workspace files written to this folder will be automatically processed by Geoscience ANALYST.
+- **workspace_geoh5** ``str`` the absolute path to the current workspace (if previously saved) being used
+- **geoh5** ``str`` the absolute path to the geoh5 written containing all the objects of the workspace within the parameters of the **ui.json**. One only needs to use this workspace along with the JSON file to access the objects with geoh5py.
+
+
+Parameters available for all ui.json objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The following members are available to all input parameters in the ui.json schema.
 
 label ``str``
     Required string describing parameter. A colon will automatically be added within Geoscience ANALYST, so this should be omitted.
 value ``str``, ``int``, ``bool`` , or ``float``
-    This require member takes a different form, including empty, depending on the :ref:`parameter type <json_param_examples>`. The value is updated when written from Geoscience ANALYST.
+    This required member takes a different form, including empty, depending on the :ref:`parameter type <json_param_examples>`. The value is updated when written from Geoscience ANALYST.
```

Comment:
`empty` implies nothing, which is not allowed generally in json. Guess you meant ""?